src/api/fetchData.js

```diff
@@ -2,24 +2,31 @@ import { useState, useEffect } from 'react';
 
 export default function fetchData(url) {
     const [data, setData] = useState([]);
-    const [loading, setLoading] = useState(null);
+    const [loading, setLoading] = useState(true);
     const [error, setError] = useState('');
 
+    const rawgioApiKey = import.meta.env.VITE_API_KEY;
+    const requestOptions = {
+        headers: {
+            'Target-URL': 'https://rawg.io',
+            Authorization: '', // cors-proxy expects this attribute, otherwise it fails
+        },
+    };
+
     useEffect(() => {
-        setLoading(true);
-        try {
-            fetch(url)
-                .then((res) => res.json())
-                .then((data) => setData(data));
-        } catch (e) {
-            throw new Error(
-                setError(
-                    `Cannot fetch the data, error with status ${response.status}`
-                )
-            );
-        } finally {
-            setLoading(false);
-        }
+        fetch(`${url}?key=${rawgioApiKey}`, requestOptions)
+            .then((res) => {
+                if (res.status >= 400) {
```

Comment:
I think it would be better to check for: `if (!res.ok) { ... }`
since it would cover any case that's not a 2XX